include/circt/Dialect/LTL/LTLOps.td

```diff
@@ -104,6 +104,21 @@ def DelayOp : LTLOp<"delay", [Pure]> {
   }];
 }
 
+def PastOp : LTLOp<"past", [Pure]> {
+  let arguments = (ins
+    LTLAnySequenceType:$input,
```

Comment:
What are the semantics of `ltl.past` when used on a sequence instead of a boolean? Does that mean the sequence must have finished matching in the previous cycle? And how would we print an `ltl.past` with a sequence input back as Verilog?

If I remember correctly, the `$past` function on Verilog only operates on booleans. In that case it might make sense to make `ltl.past` operate on just `i1` for now. It's always easier to start with an overconstrained op and relax later, than figuring out semantics later 😃

Reply:
That's a fair point, I'll limit it to i1 for this PR!